library/Core/src/Juvix/Core/Translate.hs

```diff
@@ -37,6 +37,9 @@ hrToIRWith pats term =
     |> exec pats mempty
     |> fst
 
+-- | @hrToIR'@ transforms an HR term into an IR term. The is achieved
+-- by pushing binder's variables to the stack, then getting the index
+-- when names occur for the proper De-bruijn index
```

Comment:
I think this is correct, but could be made more clear by pointing out that `hrToIR'` does the pushing and `hrElimToIR'` does the getting.  Based on the comment, I first looked into `hrToIR'` for where it ever got the index, and I couldn't find where -- it took me a little while to think of reading down, and found that that happens in `hrElimToIR'`'s call to `lookupName`, in its `HR.Var` case.

So I'd emphasize that the two mutually recursive functions split the functionality, analogously to other operations:  `Term`s have their types checked and `Elim`s have their types synthesized, for example; here, `hrToIR'` pushes binders onto the stack (using `withName`) and `hrElimToIR'` turns them into de Bruijn indices (using `lookupName`).

At least, I think that's what's happening :)